float32 desired_speed
---
float32 previous_speed
float32 new_speed
bool stalled
